drivers/sensor/lis2dh/lis2dh.c

```diff
@@ -321,6 +321,16 @@ int lis2dh_init(struct device *dev)
 		return status;
 	}
 
+#ifdef LIS2DH_HAS_SDO_SA0_PULL_UP
+	status = lis2dh_reg_field_update(dev, LIS2DH_REG_CTRL0,
+					 LIS2DH_SDO_PU_DISC_SHIFT,
+					 LIS2DH_SDO_PU_DISC_MASK, 1);
+	if (status < 0) {
+		LOG_ERR("Failed to disable SDO/SA0 pull-up.");
+		return status;
+	}
+#endif
+
```

Comment:
Why always disabling the pullup? Shouldn't it be done only selectively depending from board to board?

Reply:
There might be boards which actually requires the internal pullup to be set. So, I'm wondering if need to keep it configurable in Kconfig.

Reply:
My initial idea was to only disable it when the I2C address is `(addr & 0x1 == 0)`, i.e. 0x18 (which is the case on my HW). However this check would be valid only for I2C devices, not SPI. I think that SPI might have the same issue with power consumption, so it is good to disable that feature there as well. So lets consider those cases:
1) I2C addr is 0x18: we want to disable SDO/SA0 for sure, because we have pull-down on the board and a pull-up internally in the accelerometer,
2) I2C addr is 0x19: assuming that address is assigned during accelerometer power on, then we can either keep the pull-ups or disable them,
3) SPI: driving SDO pin will conflict a bit with pull-up on that line, which means that we might have increased power consumption on driving SDO line low.

Based on those I decided to always disable pull-up to simplify the code.

Reply:
And why not leaving it configurable in Kconfig and disabled by default?
In this case you leave the possibility to enable it in case any board requires it.
It is just brain storming...

Reply:
Because it is not a software feature which we might want to enable or disable, such as triggering events on drdy. If we want to keep it configurable, then dts seems a better place, because there we actually describe how the hardware looks like. So we might introduce a `disable-sdo-sda0-pull-up` flag to control that behavior.

However from the 3 cases that I described above I am only a bit worried about case 2, i.e. whether the address is latched on applying power or it is rather "runtime" configuration. If it is latched, then we do not need to expose any configuration option to the user at all (either Kconfig or dts). If it is NOT latched, then we can add a condition `if (I2C && (addr & 0x1))`.

The reason I don't want to keep it configurable is simplicity for the end user. It is much easier to just enable the driver, configure in dts and expect it to perform well. If someone will need to examine which IC consumes too much power when the board has 10 sensors, then lot of configuration options and dts flags would need to be investigated before finding the one we want to introduce in this PR. And the last thing... I don't see any value of keeping it configurable, as there are limited use cases we can cover without exposing such option to user.